envbuilder.go

```diff
@@ -1137,43 +971,6 @@ func createPostStartScript(path string, postStartCommand devcontainer.LifecycleS
 	return nil
 }
 
-// OptionsFromEnv returns a set of options from environment variables.
-func OptionsFromEnv(getEnv func(string) (string, bool)) Options {
```

Comment:
It feels like some subtle bugs could arise from removing this, and relying on serpent's handling of envs. Consider [Hyrum's Law](https://www.hyrumslaw.com/); if we're changing the way envs are handled then it may break someone's usage; I think we should have a set of tests covering this.

Reply:
This could be in a follow-up PR.

Reply:
I second this; there may be cases that our existing integration tests are not catching.

Reply:
Since this is public API, it should probably be made backwards compatible anyway. We may add a deprecation notice on this if we don't want it used anymore:

https://go.dev/wiki/Deprecated

Reply:
Envbuilder is still 0.x so there are no compatibility guarantees right now. If we want to make breaking API changes, now is the time.

Reply:
That's fair 👍🏻.

Reply:
Ok, since @johnstcn and @dannykopping have brought up the concern about how Serpent handles env vars, I will add a test for it 💯 😆. Would this be good enough or do I need to take any extra action?

Reply:
Marking this as resolved. I'm going to post a comment with all the follow-ups later.